libs/librepcb/workspace/settings/workspacesettingsdialog.ui

```diff
@@ -280,6 +280,143 @@
+      <layout class="QVBoxLayout" name="verticalLayout_2">
+       <item>
+        <widget class="QGroupBox" name="pdfBox">
+         <property name="title">
+          <string>PDF Reader</string>
+         </property>
+         <layout class="QGridLayout" name="gridLayout">
+          <item row="0" column="0">
+           <widget class="QLabel" name="label_13">
+            <property name="text">
+             <string>Reader to use</string>
+            </property>
+           </widget>
+          </item>
+          <item row="0" column="2">
+           <layout class="QHBoxLayout" name="pdfReaderVLayout">
+            <item>
+             <widget class="QRadioButton" name="pdfDefaultRadioBtn">
+              <property name="text">
+               <string>System default</string>
+              </property>
+              <property name="checked">
+               <bool>true</bool>
+              </property>
+              <attribute name="buttonGroup">
+               <string notr="true">pdfReaderGroup</string>
+              </attribute>
+             </widget>
+            </item>
+            <item>
+             <widget class="QRadioButton" name="pdfCustomRadioBtn">
+              <property name="text">
+               <string>Run custom command</string>
+              </property>
+              <attribute name="buttonGroup">
+               <string notr="true">pdfReaderGroup</string>
+              </attribute>
+             </widget>
+            </item>
+           </layout>
+          </item>
+          <item row="2" column="2">
+           <layout class="QHBoxLayout" name="pdfOpenVLayout">
+            <item>
+             <widget class="QRadioButton" name="pdfOpenAlwaysRadio">
+              <property name="text">
+               <string>Always</string>
+              </property>
+              <property name="checked">
+               <bool>true</bool>
+              </property>
+              <attribute name="buttonGroup">
+               <string notr="true">pdfOpenGroup</string>
+              </attribute>
+             </widget>
+            </item>
+            <item>
+             <widget class="QRadioButton" name="pdfOpenNeverRadio">
+              <property name="text">
+               <string>Never</string>
+              </property>
+              <attribute name="buttonGroup">
+               <string notr="true">pdfOpenGroup</string>
+              </attribute>
+             </widget>
+            </item>
+            <item>
+             <widget class="QRadioButton" name="pdfOpenAskRadio">
+              <property name="text">
+               <string>Ask every time</string>
+              </property>
+              <attribute name="buttonGroup">
+               <string notr="true">pdfOpenGroup</string>
+              </attribute>
+             </widget>
+            </item>
+           </layout>
+          </item>
+          <item row="1" column="2">
+           <widget class="QLineEdit" name="pdfCustomCmdEdit">
```

Comment:
For this line edit, it would be good to set following properties:
- `maxLength` to something reasonable, e.g. 200 or so (there were already strange behavior of the GUI reported when pasting extremely long texts into such fields, this can be prevented by limiting its length).
- `placeholderText` to show a usage example like `e.g. "evince {{FILEPATH}}"` to make it more intuitive.
- `clearButtonEnabled` to `true` (sometimes useful, and shouldn't hurt :wink:)

Reply:
Done. Though I was wondering if I should put a file picker button on the right of the TextEdit, because on Windows "to run a custom command" is pretty much synonymous for pick an exe somewhere.

Reply:
Yeah actually I was also thinking about that :wink: Probably such s button to pick an executable would indeed be useful. Also because you could then auto-fill the TextEdit with the choosen path and just append "{{FILEPATH}}", so for typical applications the user don't have to specify the arguments manually...

If you'd like to add such a button - perfect! If not - no problem, I'd still merge this PR :)

Reply:
Done